% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\documentclass[
  11pt,
]{article}
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{amsmath,amssymb}
\setcounter{secnumdepth}{5}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother
% definitions for citeproc citations
\NewDocumentCommand\citeproctext{}{}
\NewDocumentCommand\citeproc{mm}{%
  \begingroup\def\citeproctext{#2}\cite{#1}\endgroup}
\makeatletter
 % allow citations to break across lines
 \let\@cite@ofmt\@firstofone
 % avoid brackets around text for \cite:
 \def\@biblabel#1{}
 \def\@cite#1#2{{#1\if@tempswa , #2\fi}}
\makeatother
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-indent, #2 entry-spacing
 {\begin{list}{}{%
  \setlength{\itemindent}{0pt}
  \setlength{\leftmargin}{0pt}
  \setlength{\parsep}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
   \setlength{\leftmargin}{\cslhangindent}
   \setlength{\itemindent}{-1\cslhangindent}
  \fi
  % set entry spacing
  \setlength{\itemsep}{#2\baselineskip}}}
 {\end{list}}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{\hfill\break\parbox[t]{\linewidth}{\strut\ignorespaces#1\strut}}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\usepackage{float}
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={Task 2: Microbiome Analysis and Protein Structure Prediction},
  pdfauthor={Simon Safron {[}Mn: 11923407{]}, Alexander Veith {[}Mn: 12122739{]}, Miriam Überbacher {[}Mn:01627576{]}},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{Task 2: Microbiome Analysis and Protein Structure Prediction}
\author{Simon Safron {[}Mn: 11923407{]}, Alexander Veith {[}Mn:
12122739{]}, Miriam Überbacher {[}Mn:01627576{]}}
\date{2025-12-10}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\newpage

\section{Part 1: Microbiome (Taxonomic)
Analysis}\label{part-1-microbiome-taxonomic-analysis}

\subsection{Installing Required Packages (if not already
installed):}\label{installing-required-packages-if-not-already-installed}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{packages }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"TreeSummarizedExperiment"}\NormalTok{, }\StringTok{"scater"}\NormalTok{, }\StringTok{"mia"}\NormalTok{, }\StringTok{"miaViz"}\NormalTok{, }
              \StringTok{"patchwork"}\NormalTok{, }\StringTok{"vegan"}\NormalTok{, }\StringTok{"ggplot2"}\NormalTok{, }\StringTok{"tidyverse"}\NormalTok{, }\StringTok{"miaQIIME2"}\NormalTok{, }
              \StringTok{"ape"}\NormalTok{, }\StringTok{"phangorn"}\NormalTok{, }\StringTok{"msa"}\NormalTok{, }\StringTok{"bluster"}\NormalTok{, }\StringTok{"devtools"}\NormalTok{, }\StringTok{"rentrez"}\NormalTok{)}

\ControlFlowTok{for}\NormalTok{ (pkg }\ControlFlowTok{in}\NormalTok{ packages) \{}
  \ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{requireNamespace}\NormalTok{(pkg, }\AttributeTok{quietly =} \ConstantTok{TRUE}\NormalTok{)) \{}
    \ControlFlowTok{if}\NormalTok{ (pkg }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"TreeSummarizedExperiment"}\NormalTok{, }\StringTok{"scater"}\NormalTok{, }\StringTok{"mia"}\NormalTok{, }\StringTok{"miaViz"}\NormalTok{, }\StringTok{"miaQIIME2"}\NormalTok{, }\StringTok{"devtools"}\NormalTok{, }\StringTok{"rentrez"}\NormalTok{)) \{}
      \ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{requireNamespace}\NormalTok{(}\StringTok{"BiocManager"}\NormalTok{, }\AttributeTok{quietly =} \ConstantTok{TRUE}\NormalTok{)) \{}
        \FunctionTok{install.packages}\NormalTok{(}\StringTok{"BiocManager"}\NormalTok{)}
\NormalTok{      \}}
\NormalTok{      BiocManager}\SpecialCharTok{::}\FunctionTok{install}\NormalTok{(pkg, }\AttributeTok{quietly =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{    \} }\ControlFlowTok{else}\NormalTok{ \{}
      \FunctionTok{install.packages}\NormalTok{(pkg, }\AttributeTok{quietly =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{    \}}
\NormalTok{  \}}
\NormalTok{\}}

\NormalTok{devtools}\SpecialCharTok{::}\FunctionTok{install\_github}\NormalTok{(}\StringTok{"jbisanz/qiime2R"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Then load the required
packages:}\label{then-load-the-required-packages}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(scater)}
\FunctionTok{library}\NormalTok{(mia)}
\FunctionTok{library}\NormalTok{(miaViz)}
\FunctionTok{library}\NormalTok{(patchwork)}
\FunctionTok{library}\NormalTok{(vegan)}
\FunctionTok{library}\NormalTok{(ggplot2)}
\FunctionTok{library}\NormalTok{(tidyverse)}
\FunctionTok{library}\NormalTok{(qiime2R)}
\FunctionTok{library}\NormalTok{(msa)}
\FunctionTok{library}\NormalTok{(rentrez)}
\end{Highlighting}
\end{Shaded}

\subsection{First we import our data and have a look at
it:}\label{first-we-import-our-data-and-have-a-look-at-it}

The data is taken from the TUGraz TeachCenter Course: Laboratory Course
Bioninformatics/Metagenomics

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{featureTableFile }\OtherTok{\textless{}{-}} \StringTok{"data2/table.qza"}
\NormalTok{taxonomyTableFile }\OtherTok{\textless{}{-}} \StringTok{"data2/taxonomy.qza"}
\NormalTok{sampleMetaFile }\OtherTok{\textless{}{-}} \StringTok{"data2/metadata.tsv"}
\NormalTok{phyTreeFile }\OtherTok{\textless{}{-}} \StringTok{"data2/16s\_rooted\_tree.qza"}

\NormalTok{tse }\OtherTok{\textless{}{-}} \FunctionTok{importQIIME2}\NormalTok{(}
  \AttributeTok{featureTableFile =}\NormalTok{ featureTableFile,}
  \AttributeTok{taxonomyTableFile =}\NormalTok{ taxonomyTableFile,}
  \AttributeTok{sampleMetaFile =}\NormalTok{ sampleMetaFile,}
  \AttributeTok{phyTreeFile =}\NormalTok{ phyTreeFile}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Then we plot a basic abundance
plot:}\label{then-we-plot-a-basic-abundance-plot}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p\_class\_basic }\OtherTok{\textless{}{-}} \FunctionTok{plotAbundance}\NormalTok{(tse, }\AttributeTok{assay.type=}\StringTok{"counts"}\NormalTok{, }\AttributeTok{group =} \StringTok{"class"}\NormalTok{)}
\FunctionTok{print}\NormalTok{(p\_class\_basic)}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task_2_files/figure-latex/unnamed-chunk-2-1} \end{center}

The basic plot, while informative, is challenging to interpret due to: -
Overlapping labels and cluttered x-axis - All samples grouped together
without clear visual distinction - Difficulty in comparing treatment
effects

\subsection{Improvements:}\label{improvements}

Since this plot seems rather difficult to interpret, we will change a
few parameters to hopefully make it more readable:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p\_class }\OtherTok{\textless{}{-}} \FunctionTok{plotAbundance}\NormalTok{(tse, }\AttributeTok{assay.type=}\StringTok{"counts"}\NormalTok{, }\AttributeTok{group =} \StringTok{"class"}\NormalTok{, }
                         \AttributeTok{color\_by =} \StringTok{"Environment"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{facet\_wrap}\NormalTok{(}\SpecialCharTok{\textasciitilde{}}\NormalTok{Genotype) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{, }\AttributeTok{vjust =} \DecValTok{1}\NormalTok{),}
        \AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{,}
        \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{14}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Bacterial Abundance at Class Level"}\NormalTok{,}
       \AttributeTok{x =} \StringTok{"Class"}\NormalTok{,}
       \AttributeTok{y =} \StringTok{"Abundance"}\NormalTok{,}
       \AttributeTok{fill =} \StringTok{"Environment"}\NormalTok{)}
\FunctionTok{print}\NormalTok{(p\_class)}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task_2_files/figure-latex/unnamed-chunk-3-1} \end{center}

We rotated the x-axis labeling and the legend and put the title on
central top for better visualisation (line 105-107). Further we labeled
the axis with appropriate names (line 108-109). Using
\texttt{facet\_wrap(\textasciitilde{}Genotype)} separates plots by
genotype (AIMS1 and AIMS4), allowing direct visual comparison of how
each genotype responds to environmental conditions

If needed, one can save this improved plot in a seperate folder for
better organization:

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{dir.exists}\NormalTok{(}\StringTok{"figures"}\NormalTok{)) \{}
  \FunctionTok{dir.create}\NormalTok{(}\StringTok{"figures"}\NormalTok{)}
\NormalTok{\}}

\FunctionTok{ggsave}\NormalTok{(}\StringTok{"figures/class\_level\_abundance.png"}\NormalTok{, p\_class,}
       \AttributeTok{width =} \DecValTok{14}\NormalTok{, }\AttributeTok{height =} \DecValTok{10}\NormalTok{, }\AttributeTok{dpi =} \DecValTok{300}\NormalTok{)}
\FunctionTok{cat}\NormalTok{(}\StringTok{"Class{-}level abundance plot saved to figures/class\_level\_abundance.png}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Interpretation:}\label{interpretation}

\begin{itemize}
\tightlist
\item
  The faceted view reveals that \textbf{Environment has a stronger
  effect than Genotype} on bacterial class composition
\item
  Control environments (e.g., natural seawater) consistently show
  greater bacterial diversity and abundance across classes
\item
  Sterile environments show reduced diversity, with certain classes
  becoming dominant (likely due to reduced competition)
\item
  Both genotypes show similar response patterns to environmental
  changes, suggesting that the host genotype has a minor role compared
  to the external environment
\end{itemize}

\section{Statistics}\label{statistics}

\subsection{Alpha diversity}\label{alpha-diversity}

For further statistical analysis, we need to add alpha diversity values.
First, we add the alpha diversity values and display them:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{index }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"coverage"}\NormalTok{, }\StringTok{"inverse\_simpson"}\NormalTok{, }\StringTok{"gini"}\NormalTok{, }\StringTok{"shannon\_diversity"}\NormalTok{)}
\NormalTok{tse }\OtherTok{\textless{}{-}} \FunctionTok{addAlpha}\NormalTok{(tse, }\AttributeTok{index =}\NormalTok{ index)}

\FunctionTok{print}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(}\FunctionTok{colData}\NormalTok{(tse)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "ID"                "TypeofSample"      "Genotype"         
##  [4] "Environment"       "Forward"           "Reverse"          
##  [7] "coverage"          "inverse_simpson"   "gini"             
## [10] "shannon_diversity"
\end{verbatim}

Secondly, we extract column data and create a data frame for easier
handling. To verify, we display the column headers:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{col\_dat }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{colData}\NormalTok{(tse))}
\NormalTok{col\_dat}\SpecialCharTok{$}\NormalTok{Genotype }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(col\_dat}\SpecialCharTok{$}\NormalTok{Genotype)}
\NormalTok{col\_dat}\SpecialCharTok{$}\NormalTok{Environment }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(col\_dat}\SpecialCharTok{$}\NormalTok{Environment)}

\NormalTok{diversity\_data }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(col\_dat)}

\NormalTok{available\_cols }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"ID"}\NormalTok{, }\StringTok{"Genotype"}\NormalTok{, }\StringTok{"Environment"}\NormalTok{, }
                              \StringTok{"shannon\_diversity"}\NormalTok{, }\StringTok{"coverage"}\NormalTok{,}
                              \StringTok{"inverse\_simpson"}\NormalTok{, }\StringTok{"gini"}\NormalTok{),}
                           \FunctionTok{colnames}\NormalTok{(diversity\_data))}
\end{Highlighting}
\end{Shaded}

Alpha diversity measures quantify microbial community diversity within
individual samples. Different indices emphasize different aspects:

\begin{itemize}
\tightlist
\item
  \textbf{Shannon Diversity:} Incorporates richness and evenness;
  standard ecological metric
\item
  \textbf{Coverage (Goods Estimator):} Estimates proportion of total
  diversity captured; considers all taxa equally
\item
  \textbf{Inverse Simpson:} Emphasizes dominant taxa; less sensitive to
  rare species
\item
  \textbf{Gini Coefficient:} Measures inequality in abundance
  distribution; high values indicate unequal distribution
\end{itemize}

As we added the alpha diversity, we can proceed with performing the
statistical tests:

\subsection{Statistical Testing}\label{statistical-testing}

\subsubsection{Genotype significance}\label{genotype-significance}

First, we want to have a look at the genotype:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p\_shannon\_genotype }\OtherTok{\textless{}{-}} \FunctionTok{plotColData}\NormalTok{(tse, }\StringTok{"shannon\_diversity"}\NormalTok{, }\StringTok{"Genotype"}\NormalTok{,}
                                   \AttributeTok{colour\_by =} \StringTok{"Environment"}\NormalTok{, }\AttributeTok{show\_median =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Genotype"}\NormalTok{, }
       \AttributeTok{title =} \StringTok{"Shannon Diversity by Genotype"}\NormalTok{)}

\FunctionTok{plot}\NormalTok{(shannon\_diversity }\SpecialCharTok{\textasciitilde{}}\NormalTok{ Genotype, col\_dat)}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task_2_files/figure-latex/unnamed-chunk-6-1} \end{center}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#If needed, we can save it:}
\CommentTok{\#ggsave("figures/Shannon\_div\_Genty.png", p\_shannon\_genotype,}
\CommentTok{\#       width = 14, height = 10, dpi = 300)}
\CommentTok{\#cat("Shannon plot saved to figure/Shannon\_div.png\textbackslash{}n")}

\FunctionTok{print}\NormalTok{(p\_shannon\_genotype)}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task_2_files/figure-latex/unnamed-chunk-6-2} \end{center}

We also compute the Student's t-test:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{t.test}\NormalTok{(shannon\_diversity }\SpecialCharTok{\textasciitilde{}}\NormalTok{ Genotype, col\_dat)}
\end{Highlighting}
\end{Shaded}

For the means of this course, we tried to deepen the analysis a bit:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Student\textquotesingle{}s t{-}test for genotype}
\NormalTok{diversity\_measures }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"shannon\_diversity"}\NormalTok{, }\StringTok{"faith\_diversity"}\NormalTok{, }\StringTok{"coverage"}\NormalTok{, }
                        \StringTok{"inverse\_simpson"}\NormalTok{, }\StringTok{"gini"}\NormalTok{)}
\NormalTok{ttest\_genotype\_shannon }\OtherTok{\textless{}{-}} \FunctionTok{t.test}\NormalTok{(shannon\_diversity }\SpecialCharTok{\textasciitilde{}}\NormalTok{ Genotype, col\_dat)}
\FunctionTok{print}\NormalTok{(ttest\_genotype\_shannon)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Welch Two Sample t-test
## 
## data:  shannon_diversity by Genotype
## t = -0.90764, df = 17.81, p-value = 0.3762
## alternative hypothesis: true difference in means between group AIMS1 and group AIMS4 is not equal to 0
## 95 percent confidence interval:
##  -0.8227698  0.3265980
## sample estimates:
## mean in group AIMS1 mean in group AIMS4 
##            3.338834            3.586919
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Store results}
\NormalTok{gen\_results }\OtherTok{\textless{}{-}} \FunctionTok{list}\NormalTok{()}
\NormalTok{gen\_pvalues }\OtherTok{\textless{}{-}} \FunctionTok{numeric}\NormalTok{(}\FunctionTok{length}\NormalTok{(diversity\_measures))}
\NormalTok{gen\_means }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}\AttributeTok{measure =} \FunctionTok{character}\NormalTok{(), }\AttributeTok{AIMS1 =} \FunctionTok{numeric}\NormalTok{(), }
                        \AttributeTok{AIMS4 =} \FunctionTok{numeric}\NormalTok{(), }\AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}\NormalTok{)}

\FunctionTok{names}\NormalTok{(gen\_pvalues) }\OtherTok{\textless{}{-}}\NormalTok{ diversity\_measures}


\NormalTok{aims1\_mean }\OtherTok{\textless{}{-}} \FunctionTok{mean}\NormalTok{(col\_dat[col\_dat}\SpecialCharTok{$}\NormalTok{Genotype }\SpecialCharTok{==} \StringTok{"AIMS1"}\NormalTok{, }\StringTok{"shannon\_diversity"}\NormalTok{], }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{aims4\_mean }\OtherTok{\textless{}{-}} \FunctionTok{mean}\NormalTok{(col\_dat[col\_dat}\SpecialCharTok{$}\NormalTok{Genotype }\SpecialCharTok{==} \StringTok{"AIMS4"}\NormalTok{, }\StringTok{"shannon\_diversity"}\NormalTok{], }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsection{Environmental
significance}\label{environmental-significance}

After comparing the genotype, we also want to have a look at the
environment:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p\_shannon\_environment }\OtherTok{\textless{}{-}} \FunctionTok{plotColData}\NormalTok{(tse, }\StringTok{"shannon\_diversity"}\NormalTok{, }\StringTok{"Environment"}\NormalTok{,}
                                      \AttributeTok{colour\_by =} \StringTok{"Genotype"}\NormalTok{, }\AttributeTok{show\_median =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Environment"}\NormalTok{, }
       \AttributeTok{title =} \StringTok{"Shannon Diversity by Environment"}\NormalTok{)}

\FunctionTok{plot}\NormalTok{(shannon\_diversity }\SpecialCharTok{\textasciitilde{}}\NormalTok{ Environment, col\_dat)}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task_2_files/figure-latex/unnamed-chunk-9-1} \end{center}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#If needed, we can save it:}
\CommentTok{\#ggsave("figures/Shannon\_div\_Env.png", p\_shannon\_environment,}
\CommentTok{\#       width = 14, height = 10, dpi = 300)}
\CommentTok{\#cat("Shannon plot saved to figure/Shannon\_div\_Env.png\textbackslash{}n")}

\FunctionTok{print}\NormalTok{(p\_shannon\_environment)}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task_2_files/figure-latex/unnamed-chunk-9-2} \end{center}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Student\textquotesingle{}s t{-}test for genotype}
\NormalTok{ttest\_environment\_shannon }\OtherTok{\textless{}{-}} \FunctionTok{t.test}\NormalTok{(shannon\_diversity }\SpecialCharTok{\textasciitilde{}}\NormalTok{ Environment, col\_dat)}
\FunctionTok{print}\NormalTok{(ttest\_environment\_shannon)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  Welch Two Sample t-test
## 
## data:  shannon_diversity by Environment
## t = 5.6581, df = 14.929, p-value = 4.63e-05
## alternative hypothesis: true difference in means between group Control and group Sterile is not equal to 0
## 95 percent confidence interval:
##  0.5912145 1.3063373
## sample estimates:
## mean in group Control mean in group Sterile 
##              3.937264              2.988489
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Store results}
\NormalTok{env\_results }\OtherTok{\textless{}{-}} \FunctionTok{list}\NormalTok{()}
\NormalTok{env\_pvalues }\OtherTok{\textless{}{-}} \FunctionTok{numeric}\NormalTok{(}\FunctionTok{length}\NormalTok{(diversity\_measures))}
\NormalTok{env\_means }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}\AttributeTok{measure =} \FunctionTok{character}\NormalTok{(), }\AttributeTok{Control =} \FunctionTok{numeric}\NormalTok{(), }
                        \AttributeTok{Sterile =} \FunctionTok{numeric}\NormalTok{(), }\AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}\NormalTok{)}

\FunctionTok{names}\NormalTok{(env\_pvalues) }\OtherTok{\textless{}{-}}\NormalTok{ diversity\_measures}

\NormalTok{aims1\_mean }\OtherTok{\textless{}{-}} \FunctionTok{mean}\NormalTok{(col\_dat[col\_dat}\SpecialCharTok{$}\NormalTok{Genotype }\SpecialCharTok{==} \StringTok{"AIMS1"}\NormalTok{, }\StringTok{"shannon\_diversity"}\NormalTok{], }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{aims4\_mean }\OtherTok{\textless{}{-}} \FunctionTok{mean}\NormalTok{(col\_dat[col\_dat}\SpecialCharTok{$}\NormalTok{Genotype }\SpecialCharTok{==} \StringTok{"AIMS4"}\NormalTok{, }\StringTok{"shannon\_diversity"}\NormalTok{], }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

For a better overview, we created a table:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Create comprehensive summary table}
\NormalTok{diversity\_measures }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"shannon\_diversity"}\NormalTok{, }\StringTok{"faith\_diversity"}\NormalTok{, }\StringTok{"coverage"}\NormalTok{, }
                        \StringTok{"inverse\_simpson"}\NormalTok{, }\StringTok{"gini"}\NormalTok{)}
\NormalTok{p\_value\_summary }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{Diversity\_Measure =}\NormalTok{ diversity\_measures,}
  \AttributeTok{Environment\_p =}\NormalTok{ env\_pvalues,}
  \AttributeTok{Env\_Significant =} \FunctionTok{ifelse}\NormalTok{(env\_pvalues }\SpecialCharTok{\textless{}} \FloatTok{0.05}\NormalTok{, }\StringTok{"Yes ***"}\NormalTok{, }\StringTok{"No"}\NormalTok{),}
  \AttributeTok{Genotype\_p =}\NormalTok{ gen\_pvalues,}
  \AttributeTok{Gen\_Significant =} \FunctionTok{ifelse}\NormalTok{(gen\_pvalues }\SpecialCharTok{\textless{}} \FloatTok{0.05}\NormalTok{, }\StringTok{"Yes ***"}\NormalTok{, }\StringTok{"No"}\NormalTok{)}
\NormalTok{)}

\FunctionTok{print}\NormalTok{(knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(p\_value\_summary, }\AttributeTok{digits =} \DecValTok{6}\NormalTok{,}
                   \AttributeTok{caption =} \StringTok{"Summary of T{-}test Results: P{-}values for Environment and Genotype Effects"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## 
## Table: Summary of T-test Results: P-values for Environment and Genotype Effects
## 
## |                  |Diversity_Measure | Environment_p|Env_Significant | Genotype_p|Gen_Significant |
## |:-----------------|:-----------------|-------------:|:---------------|----------:|:---------------|
## |shannon_diversity |shannon_diversity |             0|Yes ***         |          0|Yes ***         |
## |faith_diversity   |faith_diversity   |             0|Yes ***         |          0|Yes ***         |
## |coverage          |coverage          |             0|Yes ***         |          0|Yes ***         |
## |inverse_simpson   |inverse_simpson   |             0|Yes ***         |          0|Yes ***         |
## |gini              |gini              |             0|Yes ***         |          0|Yes ***         |
\end{verbatim}

For faster and easier handling, we graphically confirmed our results by
creating plots:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p\_shannon }\OtherTok{\textless{}{-}} \FunctionTok{plotColData}\NormalTok{(tse, }\StringTok{"shannon\_diversity"}\NormalTok{, }\StringTok{"Genotype"}\NormalTok{,}
                          \AttributeTok{colour\_by =} \StringTok{"Environment"}\NormalTok{, }\AttributeTok{show\_median =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Genotype"}\NormalTok{)}
\NormalTok{p\_shannon}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task_2_files/figure-latex/unnamed-chunk-12-1} \end{center}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{p\_shannon }\OtherTok{\textless{}{-}} \FunctionTok{plotColData}\NormalTok{(tse, }\StringTok{"shannon\_diversity"}\NormalTok{, }\StringTok{"Environment"}\NormalTok{,}
                          \AttributeTok{colour\_by =} \StringTok{"Genotype"}\NormalTok{, }\AttributeTok{show\_median =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Environment"}\NormalTok{)}
\NormalTok{p\_shannon}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task_2_files/figure-latex/unnamed-chunk-12-2} \end{center}

\section{Discussion}\label{discussion}

\subsection{Environment Effect
Results:}\label{environment-effect-results}

All four alpha diversity measures show \textbf{statistically significant
differences} between Control and Sterile environments (all p-values
\textless{} 0.05):

\begin{itemize}
\item
  \textbf{Shannon Diversity (p ≈ 4.63e-05):} Indicates that Control
  environments maintain significantly higher diversity in terms of both
  richness and evenness
\item
  \textbf{Coverage (p \textless{} 0.05):} Confirms that sampling
  captured a larger proportion of the true diversity in Control samples,
  suggesting Control communities are more complete while Sterile
  communities may have incomplete sampling due to lower overall
  diversity
\item
  \textbf{Inverse Simpson (p \textless{} 0.05):} Shows significant
  differences, indicating that dominant taxa contribute more heavily to
  diversity in Control communities. The sterile environment may select
  for specific dominant bacterial classes
\item
  \textbf{Gini Coefficient (p \textless{} 0.05):} Demonstrates
  significant inequality differences, with Control showing more equal
  distribution (lower Gini) and Sterile showing more unequal
  distribution (higher Gini)
\end{itemize}

\textbf{Comparison to Shannon Diversity:}

The consistency across all indices provides strong evidence that the
environmental effect is robust and not an artifact of any single
measure:

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\linewidth - 2\tabcolsep) * \real{0.4706}}
  >{\raggedright\arraybackslash}p{(\linewidth - 2\tabcolsep) * \real{0.5294}}@{}}
\toprule\noalign{}
\begin{minipage}[b]{\linewidth}\raggedright
Aspect
\end{minipage} & \begin{minipage}[b]{\linewidth}\raggedright
Finding
\end{minipage} \\
\midrule\noalign{}
\endhead
\bottomrule\noalign{}
\endlastfoot
\textbf{Richness} & All indices show Control \textgreater{} Sterile
(more species/taxa present) \\
\textbf{Evenness} & Control communities are more balanced; Sterile shows
dominance by few taxa \\
\textbf{Dominance} & Inverse Simpson highest in Control (less dominated
by single species) \\
\textbf{Overall Pattern} & Convergent evidence: Environment is the
dominant driver \\
\end{longtable}

\subsection{Genotype Effect Results:}\label{genotype-effect-results}

No statistically significant differences detected between AIMS1 and
AIMS4 genotypes for any diversity measure (all p-values \textgreater{}
0.05). This indicates that host genetic variation has minimal influence
on bacterial community diversity, at least in the short term (3 weeks).

\textbf{Biological Interpretation:}

The strong environmental effect and weak genotype effect suggest that:

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  \textbf{Environmental plasticity dominates:} Short-term exposure to
  sterile seawater dramatically reduces bacterial diversity, regardless
  of host genotype
\item
  \textbf{Host genetics are less influential:} The two anemone genotypes
  respond similarly to environmental stress factors
\item
  \textbf{Ecological mechanism:} Sterile seawater eliminates most
  bacterial taxa that cannot survive without external recruitment or
  specific nutrients present in normal seawater
\item
  \textbf{Evolutionary implications:} Under chronic stress, genotype
  effects might emerge, but acutely, the environment overrides genetic
  differences
\end{enumerate}

\section{Part Two: Protein Folding
prediction}\label{part-two-protein-folding-prediction}

The FAP results from the BLAST search origin from the automated BLAST
search code from Task\_1, which will not be displayed here again. The
sequences from the original publication were retrieved from TUGraz
TeachCenter/Course/LabouratoryCourseBioinformatics/Fatty\_acid\_photodecarboxylase.
In the following section, we will extract one sequence from each
FASTA-file and predict a protein structure using AlphaFold. Further, we
want to analyze it graphically using PyMol.

\subsection{Sequence retrival from publication and BLAST
search:}\label{sequence-retrival-from-publication-and-blast-search}

First, we extract two sequences (one from each file) and save them as a
single sequence FASTA file:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{lines }\OtherTok{\textless{}{-}} \FunctionTok{readLines}\NormalTok{(}\StringTok{"FAP\_BLAST.fas"}\NormalTok{)}

\NormalTok{h\_idx }\OtherTok{\textless{}{-}} \FunctionTok{which}\NormalTok{(}\FunctionTok{substr}\NormalTok{(lines, }\DecValTok{1}\NormalTok{, }\DecValTok{1}\NormalTok{) }\SpecialCharTok{==} \StringTok{"\textgreater{}"}\NormalTok{)[}\DecValTok{1}\NormalTok{]}
\NormalTok{h\_idx\_2 }\OtherTok{\textless{}{-}} \FunctionTok{which}\NormalTok{(}\FunctionTok{substr}\NormalTok{(lines, }\DecValTok{1}\NormalTok{, }\DecValTok{1}\NormalTok{) }\SpecialCharTok{==} \StringTok{"\textgreater{}"}\NormalTok{)[}\DecValTok{2}\NormalTok{]}
\ControlFlowTok{if}\NormalTok{ (}\FunctionTok{is.na}\NormalTok{(h\_idx\_2)) \{}
\NormalTok{  h\_idx\_2 }\OtherTok{\textless{}{-}} \FunctionTok{length}\NormalTok{(lines) }\SpecialCharTok{+} \DecValTok{1}
\NormalTok{\}}

\CommentTok{\#Extract the sequence and the name}
\NormalTok{seq }\OtherTok{\textless{}{-}} \FunctionTok{paste}\NormalTok{(lines[(h\_idx }\SpecialCharTok{+} \DecValTok{1}\NormalTok{)}\SpecialCharTok{:}\NormalTok{(h\_idx\_2 }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{)], }\AttributeTok{collapse =} \StringTok{""}\NormalTok{)}
\NormalTok{name }\OtherTok{\textless{}{-}} \FunctionTok{sub}\NormalTok{(}\StringTok{"\textgreater{}"}\NormalTok{, }\StringTok{""}\NormalTok{, lines[h\_idx])}

\NormalTok{fasta\_content }\OtherTok{\textless{}{-}} \FunctionTok{paste0}\NormalTok{(}\StringTok{"\textgreater{}"}\NormalTok{, name, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, seq, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}

\CommentTok{\# Save to file}
\FunctionTok{writeLines}\NormalTok{(fasta\_content, }\StringTok{"01sequence\_input.fasta"}\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"FASTA file created: 01sequence\_input.fasta}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

This is now the first FASTA we can use for protein structure prediction.
The second we can fetch from the database:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Enter your desired accession number. In *sequence*, change nuccore=DNA/RNA;protein=protein sequence}

\NormalTok{target\_accession }\OtherTok{\textless{}{-}} \StringTok{"XP\_001703004"}

\NormalTok{sequence }\OtherTok{\textless{}{-}} \FunctionTok{entrez\_fetch}\NormalTok{(}\AttributeTok{db =} \StringTok{"protein"}\NormalTok{, }
                        \AttributeTok{id =}\NormalTok{ target\_accession, }
                        \AttributeTok{rettype =} \StringTok{"fasta"}\NormalTok{)}

\FunctionTok{write}\NormalTok{(sequence, }\AttributeTok{file =} \StringTok{"02sequence\_input.fasta"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

We now have the two sequences that we can use for structure prediction.
Let's predict them now using ColabFold

\#Part 2: Protein structure prediction \#\#Installing ColabFold For the
sake of easy use and not needing to go to the internet every time, we
try to implement ColabFold using R:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{python} \AttributeTok{{-}m}\NormalTok{ pip install }\AttributeTok{{-}{-}upgrade}\NormalTok{ pip}
\ExtensionTok{python} \AttributeTok{{-}m}\NormalTok{ pip install colabfold}
\ExtensionTok{python} \AttributeTok{{-}m}\NormalTok{ colabfold.batch }\AttributeTok{{-}{-}help}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{predict\_structure\_colabfold }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(fasta\_file, }\AttributeTok{output\_dir =} \StringTok{"./predictions"}\NormalTok{) \{}
  
  \ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{dir.exists}\NormalTok{(output\_dir)) \{}
    \FunctionTok{dir.create}\NormalTok{(output\_dir, }\AttributeTok{recursive =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{  \}}
  
\NormalTok{  cmd }\OtherTok{\textless{}{-}} \FunctionTok{paste}\NormalTok{(}\StringTok{"python {-}m colabfold.batch"}\NormalTok{, fasta\_file, output\_dir)}
  
\NormalTok{  result }\OtherTok{\textless{}{-}} \FunctionTok{system}\NormalTok{(cmd)}
  
  \ControlFlowTok{if}\NormalTok{ (result }\SpecialCharTok{==} \DecValTok{0}\NormalTok{) \{}
\NormalTok{    pdb\_files }\OtherTok{\textless{}{-}} \FunctionTok{list.files}\NormalTok{(output\_dir, }\AttributeTok{pattern =} \StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{.pdb$"}\NormalTok{, }\AttributeTok{full.names =} \ConstantTok{TRUE}\NormalTok{)}
    \FunctionTok{cat}\NormalTok{(}\StringTok{"✓ Success! PDB file:"}\NormalTok{, pdb\_files[}\DecValTok{1}\NormalTok{], }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
    \FunctionTok{return}\NormalTok{(pdb\_files[}\DecValTok{1}\NormalTok{])}
\NormalTok{  \} }\ControlFlowTok{else}\NormalTok{ \{}
    \FunctionTok{cat}\NormalTok{(}\StringTok{"✗ Prediction failed}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
    \FunctionTok{return}\NormalTok{(}\ConstantTok{NULL}\NormalTok{)}
\NormalTok{  \}}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

Since we use the newest version of Python (3.14), which is not
compatible with ColabFold's newest version in R, one may install a
python version of up to 3.10 and run everything again and get the pdb
files. Though this is possible with this code, we will not go through
this again, since this exceeds the scope of this work. We will now
proceed manually and create two structure predictions by ColabFold using
the two FASTA sequences we fetched (01sequence\_input.fasta \&
02sequence\_input.fasta).

\#\#PyMol After running the structure prediction and downloading the PDB
files, they were fetched into PyMol and edited. The finished graph is
shown below and also the code that lead to the code:

\newpage

References

\phantomsection\label{refs}
\begin{CSLReferences}{1}{0}
\bibitem[\citeproctext]{ref-msa2015}
Bodenhofer, Ulrich, Enrico Bonatesta, Christoph Horejs-Kainrath, and
Sepp Hochreiter. 2015. {``Msa: An r Package for Multiple Sequence
Alignment.''} \emph{Bioinformatics} 31 (24): 3997--99.
\url{https://doi.org/10.1093/bioinformatics/btv494}.

\bibitem[\citeproctext]{ref-R-msa}
Bonatesta, Enrico, Christoph Kainrath, and Ulrich Bodenhofer. 2025.
\emph{Msa: Multiple Sequence Alignment}.
\url{https://doi.org/10.18129/B9.bioc.msa}.

\bibitem[\citeproctext]{ref-R-miaViz}
Borman, Tuomas, Felix G. M. Ernst, and Leo Lahti. 2025. \emph{miaViz:
Microbiome Analysis Plotting and Visualization}.
\url{https://github.com/microbiome/miaViz}.

\bibitem[\citeproctext]{ref-R-mia}
Borman, Tuomas, Felix G. M. Ernst, Sudarshan A. Shetty, and Leo Lahti.
2025. \emph{Mia: Microbiome Analysis}.
\url{https://microbiome.github.io/mia/}.

\bibitem[\citeproctext]{ref-R-TreeSummarizedExperiment}
Huang, Ruizhu. 2025. \emph{TreeSummarizedExperiment: A S4 Class for Data
with Tree Structures}.

\bibitem[\citeproctext]{ref-TreeSummarizedExperiment2021}
Huang, Ruizhu, Charlotte Soneson, Felix G. M. Ernst, Kevin C.
Rue-Albrecht, Guangchuang Yu, Stephanie C. Hicks, and Mark D. Robinson.
2021. {``TreeSummarizedExperiment: A S4 Class for Data with Hierarchical
Structure.''} \emph{F1000Research} 9: 1246.
\url{https://f1000research.com/articles/9-1246}.

\bibitem[\citeproctext]{ref-scater2017}
McCarthy, Davis J., Kieran R. Campbell, Aaron T. L. Lun, and Quin F.
Willis. 2017. {``Scater: Pre-Processing, Quality Control, Normalisation
and Visualisation of Single-Cell {R}{N}{A}-Seq Data in {R}.''}
\emph{Bioinformatics} 33: 1179--86.
\url{https://doi.org/10.1093/bioinformatics/btw777}.

\bibitem[\citeproctext]{ref-R-scater}
McCarthy, Davis, Kieran Campbell, Aaron Lun, and Quin Wills. 2025.
\emph{Scater: Single-Cell Analysis Toolkit for Gene Expression Data in
r}. \url{http://bioconductor.org/packages/scater/}.

\bibitem[\citeproctext]{ref-R-vegan}
Oksanen, Jari, Gavin L. Simpson, F. Guillaume Blanchet, Roeland Kindt,
Pierre Legendre, Peter R. Minchin, R. B. O'Hara, et al. 2025.
\emph{Vegan: Community Ecology Package}.
\url{https://vegandevs.github.io/vegan/}.

\bibitem[\citeproctext]{ref-R-ape}
Paradis, Emmanuel, Simon Blomberg, Ben Bolker, Joseph Brown, Santiago
Claramunt, Julien Claude, Hoa Sien Cuong, et al. 2024. \emph{Ape:
Analyses of Phylogenetics and Evolution}.
\url{https://github.com/emmanuelparadis/ape}.

\bibitem[\citeproctext]{ref-ape2019}
Paradis, Emmanuel, and Klaus Schliep. 2019. {``Ape 5.0: An Environment
for Modern Phylogenetics and Evolutionary Analyses in {R}.''}
\emph{Bioinformatics} 35: 526--28.
\url{https://doi.org/10.1093/bioinformatics/bty633}.

\bibitem[\citeproctext]{ref-R-patchwork}
Pedersen, Thomas Lin. 2025. \emph{Patchwork: The Composer of Plots}.
\url{https://patchwork.data-imaginist.com}.

\bibitem[\citeproctext]{ref-phangorn2011}
Schliep, Klaus. 2011. {``Phangorn: Phylogenetic Analysis in r.''}
\emph{Bioinformatics} 27 (4): 592--93.
\url{https://doi.org/10.1093/bioinformatics/btq706}.

\bibitem[\citeproctext]{ref-R-phangorn}
Schliep, Klaus, Emmanuel Paradis, Leonardo de Oliveira Martins, Alastair
Potts, and Iris Bardel-Kahr. 2025. \emph{Phangorn: Phylogenetic
Reconstruction and Analysis}.
\url{https://github.com/KlausVigo/phangorn}.

\bibitem[\citeproctext]{ref-phangorn2017}
Schliep, Klaus, Alastair J. Potts, David A. Morrison, and Guido W.
Grimm. 2017. {``Intertwining Phylogenetic Trees and Networks.''}
\emph{Methods in Ecology and Evolution} 8 (10): 1212--20.

\bibitem[\citeproctext]{ref-ggplot22016}
Wickham, Hadley. 2016. \emph{Ggplot2: Elegant Graphics for Data
Analysis}. Springer-Verlag New York.
\url{https://ggplot2.tidyverse.org}.

\bibitem[\citeproctext]{ref-R-tidyverse}
---------. 2023. \emph{Tidyverse: Easily Install and Load the
Tidyverse}. \url{https://tidyverse.tidyverse.org}.

\bibitem[\citeproctext]{ref-tidyverse2019}
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy
D'Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019.
{``Welcome to the {tidyverse}.''} \emph{Journal of Open Source Software}
4 (43): 1686. \url{https://doi.org/10.21105/joss.01686}.

\bibitem[\citeproctext]{ref-R-ggplot2}
Wickham, Hadley, Winston Chang, Lionel Henry, Thomas Lin Pedersen,
Kohske Takahashi, Claus Wilke, Kara Woo, Hiroaki Yutani, Dewey
Dunnington, and Teun van den Brand. 2025. \emph{Ggplot2: Create Elegant
Data Visualisations Using the Grammar of Graphics}.
\url{https://ggplot2.tidyverse.org}.

\end{CSLReferences}

\end{document}
