---
title: "Microbiome Analysis"
author: "MOL.923 Computerlabor Bioinformatik (WS 2025)"
date: "25/11/2025"
output: html_document
editor_options:
  chunk_output_type: console
markdown:
  wrap: sentence
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
suppressPackageStartupMessages(library(mia))
suppressPackageStartupMessages(library(scater))
suppressPackageStartupMessages(library(patchwork))
suppressPackageStartupMessages(library(vegan))
suppressPackageStartupMessages(library(miaViz))
```


## Background

What is the influence of genotype (intrinsic) and environment (extrinsic) on anemone-associated bacterial communities?

### The Players

* *Exaiptasia diaphana* - a shallow-water, marine anemone that is often used in research as a model organism for corals. In this experiment, two genotypes (AIMS1 and AIMS4) of *E. diaphana* were grown in each of two different environments:  
    1. sterile seawater ***OR***  
    2. unfiltered control seawater  

* The anemone-associated bacterial communities or *microbiome* - these bacteria live on, or within *E. diaphana*, and likely consist of a combination of commensals, transients, and long-term stable members, and combined with their host, form a mutually beneficial, stable symbiosis.

### The Study
The anemone microbiome contributes to the overall health of this complex system and can evolve in tandem with the anemone host. In this data set we are looking at the impact of intrinsic and extrinsic factors on anemone microbiome composition. After three weeks in either sterile or control seawater (environment), anemones were homogenized and DNA was extracted. There are *23* samples in this data set - *5* from each anemone treatment combination (2 genotypes x 2 environments) and *3* DNA extraction blanks as controls. *This data is a subset from a larger experiment*.

Dungan AM, van Oppen MJH, and Blackall LL (2021) Short-Term Exposure to Sterile Seawater Reduces Bacterial Community Diversity in the Sea Anemone, *Exaiptasia diaphana*. *Front. Mar. Sci.* 7:599314. doi:10.3389/fmars.2020.599314 [[Full Text]](https://www.frontiersin.org/articles/10.3389/fmars.2020.599314/full).

First we load some packages
```{R}
library(scater)
library(mia)
library(miaViz)
library(patchwork)
library(vegan)
```
First of all, we import data from qiime2 into R
```{R, cache=TRUE}
featureTableFile <- "data_raw/table.qza"
taxonomyTableFile <- "data_raw/taxonomy.qza"
sampleMetaFile <- "data_raw/metadata.tsv"
phyTreeFile <- "data_raw/16s_rooted_tree.qza"

tse <- importQIIME2(
  featureTableFile = featureTableFile,
  taxonomyTableFile = taxonomyTableFile,
  sampleMetaFile = sampleMetaFile,
  phyTreeFile = phyTreeFile
)

tse
```
## Abundance

First we explore the data a bit and explore the abundance.

```{R, cache=TRUE}
plotAbundance(tse, assay.type="counts", group = "phylum")
```



##  Alpha Diversity

### Estimation

We calculate alpha diversity in terms of coverage, Shannon, inverse Simpson and Faith indices based on the counts assay. The first three indices differ from one another in how much weight they give to rare taxa: coverage considers all taxa equally important, whereas Shannon and - even more - Simpson give more importance to abundant taxa. Unlike all others, Faith index measures the phylogenetic diversity and thus requires a phylogenetic tree (stored as rowTree in the TreeSE).

```{R, cache=TRUE}
# estimate the specified indices based on the counts assay

tse <- addAlpha(tse)
```

## Visualisation

Next, we plot the four indices, with patient status on the x axis and alpha diversity on the y axis. We can also colour by cohort to check for batch effects.


```{R, cache=TRUE}
colnames(colData(tse))

# plot shannon diversity vs patient_status
p_shannon <- plotColData(tse, "shannon_diversity", "Genotype",
                          colour_by = "Environment", show_median = TRUE) +
  labs(x = "Genotype")
p_shannon



# plot faith index vs patient_status
p_faith <- plotColData(tse, "faith_diversity", "Genotype",
                          colour_by = "Environment", show_median = TRUE) +
  labs(x = "Genotype")
p_faith
```

## Testing for differences between groups


We can extract a `data.frame` from the object `tse`.  We use the factor command to define 
groups for the *Genotype* and the *Environment*.  
```{R, cache=TRUE}
col_dat <- colData(tse)
col_dat$Genotype <- factor(col_dat$Genotype)
col_dat$Environment <- factor(col_dat$Environment)
```
Now we can test if there are differences between the environments or genotype. 
First we test if there are differences between the genotype. We also plot the the data 
using a boxplot and violin plot. 
```{R, cache=TRUE}
plot(shannon_diversity ~ Genotype, col_dat)
p_shannon
wilcox.test(shannon_diversity ~ Genotype, col_dat)
t.test(shannon_diversity ~ Genotype, col_dat)
```
For the genotype at an significance level $\alpha=0.05 \%$ we cannot reject the Null hypothesis that the groups are equal. The p-value (Wilcoxon test: 0.3527 and t-test: 0.3762)  is greater than  $\alpha=0.05 \%$. This does not suggest that these groups do not differ. 
The differences might be too small to identify with the small sampling size we have.



Next we test if there is a difference between the two environments. 
```{R, cache=TRUE}
plot(shannon_diversity ~ Environment, col_dat)
p_shannon <- plotColData(tse, "shannon_diversity", "Environment",
                          colour_by = "Genotype", show_median = TRUE) +
  labs(x = "Environment")
p_shannon

wilcox.test(shannon_diversity ~ Environment, col_dat)
t.test(shannon_diversity ~ Environment, col_dat)
```

For the environment the the p-values are smaller than $\alpha=0.05 \%$ (Wilcoxon test:2.165e-05 and t-test: 4.63e-05). In this case we can reject the null hypotheses and conclude that the two groups differ. 

We can also perform an ANOVA (ANalysis Of VAriance). This allows us to test both dependent variables Environment and Genotype together. 


```{R, cache=TRUE}
my_anova <- aov(shannon_diversity ~ Environment + Genotype, col_dat)
summary(my_anova)
```
Now we can see (again), that there is a statistical significant effect for the Environment, but not for the Genotype.


## Beta Diversity


### Principal Coordinate Analysis (PCoA)

PCoA is a method to choose the dimensions of the data that explain most of the variance in the data. Distances between samples can be expressed by several ecological indices, such as Bray-Curtis and Aitchison dissimilarities.

Here, we run multi-dimensional scaling (a type of PCoA) with Bray-Curtis dissimilarity and based on the relabundance assay.


We transform the counts assay to relative abundances and store the new assay back in the TreeSE.
```{R}
tse <- transformAssay(tse,
                       method = "relabundance")
```


```{R}
tse <- runMDS(tse,
              FUN = vegan::vegdist,
              method = "bray",
              name = "Bray",
              assay.type = "relabundance")
```
We then visualise the first two dimensions.
```{R}
p1 <- plotReducedDim(tse, "Bray",
                     colour_by = "Environment",
                     shape_by = "Genotype")

p1
```
We do the same for the Unifrac distance.
```{R}
tse <- runMDS(tse,
              FUN = mia::calculateUnifrac,
              name = "Unifrac",
              tree = rowTree(tse),
              ntop = nrow(tse),
              assay.type = "counts")

p2 <- plotReducedDim(tse, "Unifrac",
                     colour_by = "Environment",
                     shape_by = "Genotype")

p2
```
